### PR TITLE
Proposed Stage Save State Mirror Fix

### PR DESCRIPTION
**Pull request by ShepDogFGC**
At the time of this comment I'm setting up a testing env to see if this actually works.

### CODE REVIEW

**Comment by asimon-1** [comment]
Looks nice! I took a look and fixed two small bugs:
1. The compiler was throwing errors when trying to convert the LuaConsts to i32 in `get_stage_offset()`. Apparently you can't dereference in match patterns, so I refactored to use a hashmap instead
2. When we aren't mirroring, we don't want to apply the offset. Added a conditional to check for that.

Not sure how best to incorporate these changes in to the PR, so see discord for my edited copy of `save_states.rs`

**Comment by ShepDogFGC** [comment]
Implemented the function w/ the hashmap and thank you for explaining the addition of the statement (makes sense, just didn't think about that lmao)

File: src/training/save_states.rs [hunk 1]
```diff
@@ -325,7 +341,7 @@ pub unsafe fn save_states(module_accessor: &mut app::BattleObjectModuleAccessor)
         KineticModule::clear_speed_all(module_accessor);
 
         let pos = Vector3f {
-            x: MIRROR_STATE * save_state.x,
+            x: MIRROR_STATE * (save_state.x - get_stage_offset(stage_id())),
```
[the remark this answers is not in the document]
Comment (ShepDogFGC) [reply src/training/save_states.rs:344]: I'm an idiot I forgot to add that LMAO

**Comment by asimon-1** [comment]
Looks like there's a few compile errors left before we can merge this:
`
error[E0423]: expected value, found struct HashMap
   --> src/training/save_states.rs:163:39
    |
163 |     let offsetse: HashMap<f32, i32> = HashMap.from([
    |                                       ^^^^^^^- help: use the path separator to refer to an item: `::`


error[E0425]: cannot find value `offsets` in this scope
   --> src/training/save_states.rs:169:6
    |
169 |     *offsets.get(&stage_id).unwrap_or(&0.0)
    |      ^^^^^^^ help: a local variable with a similar name exists: offsetse


error: aborting due to 2 previous errors
`

**Pull request closed without merging**